lib/json/ld/expansion.ex

```diff
@@ -100,26 +85,24 @@ defmodule JSON.LD.Expansion do
           expanded_property = expand_iri(key, active_context, false, true)
           # 7.2)
           # 7.3)
-          if expanded_property &&
+          if is_binary(expanded_property) &&
```

Comment:
Unless Dialyzer doesn't keep quite without this, I prefer the previous version as it follows Erlang's philosophy of "Let it crash". With that we see immediately that there's a problem in the code, whereas now it might continue and produce wrong results, whose cause is much harder to track back.

Reply:
Agree, it looks weird, but it's the hack for Dialyzer indeed..

Reply:
Ok, I've looked over the code again and with `expand_iri` also returning tuples the scenario I've described isn't that unlikely. As you've seen now the algorithms here are quite complex and as you can imagine it's a nightmare to find a root cause for wrong control flows in this.
Having gone through this a couple of times and seeing that this introduces a problematic possible control flow, I'd prefer another solution. The previous version was actually the best way I can see to handle this situation. So, can you please check if there's another way to make Dialyzer pass, eg. by ignoring. If there's really no other way, then please handle the tuple case and raise an error explicitly.

Reply:
I've added flag to ignore this case and 2 another as well where Dialyzer is not happy for some reason.